pyked/converters.py

```diff
@@ -0,0 +1,741 @@
+        properties['apparatus']['kind'] = kind
+    else:
+        raise NotImplementedError(kind + ' experiment not (yet) supported')
+
+    return properties
+
+
+def get_common_properties(root):
+    """Read common properties from root of ReSpecTh XML file.
+
+    Args:
+        root (`etree.Element`): Root of ReSpecTh XML file
+
+    Returns:
+        properties (`dict`): Dictionary with common properties
+    """
+    properties = {}
+
+    for elem in root.iterfind('commonProperties/property'):
+        name = elem.attrib['name']
+
+        if name == 'initial composition':
+            properties['composition'] = {'species': []}
+            composition_type = None
+
+            for child in elem.iter('component'):
+                spec = {}
+                spec['species-name'] = child.find('speciesLink').attrib['preferredKey']
+
+                # use InChI for unique species identifier (if present)
+                try:
+                    spec['InChI'] = child.find('speciesLink').attrib['InChI']
+                except KeyError:
+                    # TODO: add InChI validator/search
+                    print('Warning: missing InChI for species ' + spec['species-name'])
+                    pass
+
+                # amount of that species
+                spec['amount'] = [float(child.find('amount').text)]
+
+                properties['composition']['species'].append(spec)
+
+                # check consistency of composition type
+                if not composition_type:
+                    composition_type = child.find('amount').attrib['units']
+                elif composition_type != child.find('amount').attrib['units']:
+                    raise KeyError('inconsistent initial composition units')
+            assert composition_type in ['mole fraction', 'mass fraction']
+            properties['composition']['kind'] = composition_type
+
+        elif name in ['temperature', 'pressure', 'pressure rise', 'compression time']:
+            field = name.replace(' ', '-')
+            units = elem.attrib['units']
+            if units == 'Torr':
+                units = 'torr'
+            quantity = 1.0 * unit_registry(units)
+            try:
+                quantity.to(property_units[field])
+            except pint.DimensionalityError:
+                raise KeyError('Error: units incompatible for property ' + name)
+
+            properties[field] = [' '.join([elem.find('value').text, units])]
+
+        else:
+            raise KeyError('Property ' + name + ' not supported as common property.')
+
+    return properties
+
+
+def get_ignition_type(root):
+    """Gets ignition type and target.
+
+    Args:
+        root (`etree.Element`): Root of ReSpecTh XML file
+
+    Returns:
+        properties (`dict`): Dictionary with ignition type/target information
+    """
+    ignition = {}
+    elem = root.find('ignitionType')
+
+    if elem is None:
+        raise AttributeError('missing ignitionType')
+
+    try:
```

Comment:
IMHO, using `get` or `if key in dict:` is easier to read, clearer (especially `if key in dict`) and more "pythonic", whatever that means 😸  FWIW, there's also some evidence (https://stackoverflow.com/a/20308657/2449192) that try/except is slower.